test/node/bigint.test.ts

```diff
@@ -0,0 +1,102 @@
+import { BSON, BSONError } from '../register-bson';
+import { bufferFromHexArray } from './tools/utils';
+import { expect } from 'chai';
+
+describe('BSON BigInt deserialization support', function () {
+  describe('BSON.deserialize()', function () {
+    type DeserialzationOptions = {
+      useBigInt64: boolean | undefined;
+      promoteValues: boolean | undefined;
+      promoteLongs: boolean | undefined;
+    };
+    type TestTableEntry = {
+      options: DeserialzationOptions;
+      shouldThrow: boolean;
+      expectedResult: BSON.Document;
+      expectedErrorMessage: string;
+    };
+    const testSerializedDoc = bufferFromHexArray(['12', '6100', '2300000000000000']); // key 'a', value 0x23 as int64
+    const useBigInt64Values = [true, false, undefined];
+    const promoteLongsValues = [true, false, undefined];
+    const promoteValuesValues = [true, false, undefined];
+
+    const testTable = useBigInt64Values.flatMap(useBigInt64 => {
+      return promoteLongsValues.flatMap(promoteLongs => {
+        return promoteValuesValues.flatMap(promoteValues => {
+          const useBigInt64Set = useBigInt64 ?? false;
+          const promoteLongsSet = promoteLongs ?? true;
+          const promoteValuesSet = promoteValues ?? true;
+          const shouldThrow = useBigInt64Set && (!promoteValuesSet || !promoteLongsSet);
+          let expectedResult: BSON.Document;
+          if (useBigInt64Set) {
+            expectedResult = { a: 0x23n };
+          } else if (promoteLongsSet && promoteValuesSet) {
+            expectedResult = { a: 0x23 };
+          } else {
+            expectedResult = { a: new BSON.Long(0x23) };
+          }
+          const expectedErrorMessage = shouldThrow
+            ? 'Must either request bigint or Long for int64 deserialization'
+            : '';
+          return [
+            {
+              options: { useBigInt64, promoteValues, promoteLongs },
+              shouldThrow,
+              expectedResult,
+              expectedErrorMessage
+            }
+          ];
+        });
+      });
+    });
+
+    it('meta test: should test table correctly', () => {
```

Comment:
```suggestion
    it('meta test: generates 27 tests 5 error cases and 22 success cases', () => {
```

Just a bit more informational, if this were to ever change we'd change the title to go with it